### ISSUE TITLE
gx does not ignore all files that git does

### ISSUE THREAD

**Issue by wigy-opensource-developer** [whyrusleeping/gx#120]
In https://github.com/whyrusleeping/gx#ignoring-files-from-a-publish I read that gx publish ignores files that are ignored by git. An ipfs/go-ipfs repo contains around 56MB of binaries after some sharness tests are run. I had to `git clean -x` before publishing in order to avoid publishing those.

Probably gx should leave it to `git status --ignored` to check which files are not to be published?

**Comment by Kubuxu** [comment]
Right, AFAIK we use root directory only `.gitignore` but there are also ones in subdirectories.

**Comment by Stebalien** [comment]
Ideally, we'd do a clean checkout (with hardlinks).
